### DISCUSSION TITLE
Callable function as 'metric' for ShapeletForestClassifier (gRSF)

### DISCUSSION THREAD

**Discussion by lorebon** [wildboar-foundation/wildboar#86] [Ideas]
Dear Isak,

once again I come up with some ideas: would it be possible to pass a callable function for the 'metric' argument in ShapeletForestClassifier? My idea is to extend the classifier to work with other distance functions, for instance the L_1 metric.

Thank you in advance!

**Comment by isaksamsten** [comment]
Hi!

that could be possible but the performance would be horrible. 

If you are only interested in the L1 norm, you can change the metric to “manhattan”. Unfortunately, I have yet to implement the scaled counterpart. As of now only “dtw” and “euclidean” has that support (“scaled_euclidean” and “scaled_dtw”). 

You can see a full list of supported metrics in the documentation: https://wildboar.dev/1.1.X/guide/metrics.html

best,
Isak

**Reply by lorebon** [reply]
Thank you again! The Manhattan metric works fine!

At the best of your knowledge, how much is the scaling factor important? I think it should be less sensitive to outliers.

**Reply by isaksamsten** [reply]
In my experience the performance difference can be quite significant, at least for Euclidean distance.

I’m currently working on a new version, I will see if I can land both features (callable, and scaling of more metrics)
